icu4c/source/test/intltest/numbertest_range.cpp

```diff
@@ -860,6 +861,37 @@ void NumberRangeFormatterTest::toObject() {
     }
 }
 
+void NumberRangeFormatterTest::testToDecimalNumbers() {
+    IcuTestErrorCode status(*this, "testToDecimalNumbers");
+
+    LocalizedNumberRangeFormatter lnf = NumberRangeFormatter::withLocale("en")
+        .numberFormatterBoth(NumberFormatter::with().unit(USD));
+
+    // Range of numbers
+    {
+        FormattedNumberRange range = lnf.formatFormattableRange(1, 5, status);
+        assertEquals("Range: Formatted string should be as expected",
+            u"$1.00 \u2013 $5.00",
+            range.toString(status));
+        auto decimalNumbers = range.getDecimalNumbers<std::string>(status);
+        // NOTE: DecNum doesn't retain trailing zeros. Is that a problem?
```

Comment:
I don't know the DecNum spec, but this means that the FormattedNumberRange getter doesn't quite return a string with the fidelity that the getter docs promise.

Reply:
I opened a ticket and added logKnownIssue to investigate further.